bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonNetworkHandler.java

```diff
@@ -78,57 +80,76 @@ public void initialize() {
-                    InsteonBinding insteonBinding = this.insteonBinding;
-                    if (insteonBinding != null && insteonBinding.startPolling()) {
-                        pollingJob = scheduler.scheduleWithFixedDelay(() -> {
-                            insteonBinding.logDeviceStatistics();
-                        }, 0, LOG_DEVICE_STATISTICS_DELAY_IN_SECONDS, TimeUnit.SECONDS);
-
-                        insteonBinding.setIsActive(true);
-
-                        updateStatus(ThingStatus.ONLINE);
-                    } else {
-                        String msg = "Initialization failed, unable to start the Insteon bridge with the port '"
-                                + config.getPort() + "'.";
-                        logger.warn(msg);
-
-                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
-                    }
-
-                    ScheduledFuture<?> settleJob = this.settleJob;
-                    if (settleJob != null) {
-                        settleJob.cancel(false);
-                    }
-                    this.settleJob = null;
+            return;
+        }
+        insteonBinding = new InsteonBinding(this, config, serialPortManager, scheduler);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // hold off on starting to poll until devices that already are defined as things are added.
+        // wait SETTLE_TIME_IN_SECONDS to start then check every second afterwards until it has been at
+        // least SETTLE_TIME_IN_SECONDS since last device was created.
+        settleJob = scheduler.scheduleWithFixedDelay(() -> {
+            // check to see if it has been at least SETTLE_TIME_IN_SECONDS since last device was created
+            if (System.currentTimeMillis() - lastInsteonDeviceCreatedTimestamp > SETTLE_TIME_IN_SECONDS * 1000) {
+                // settle time has expired start polling
+                InsteonBinding insteonBinding = this.insteonBinding;
+                if (insteonBinding != null && insteonBinding.startPolling()) {
+                    pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                        insteonBinding.logDeviceStatistics();
+                    }, 0, LOG_DEVICE_STATISTICS_DELAY_IN_SECONDS, TimeUnit.SECONDS);
+
+                    // wait until driver is initialized before setting network to ONLINE
+                    driverInitializedJob = scheduler.scheduleWithFixedDelay(() -> {
+                        if (insteonBinding.isDriverInitialized()) {
+                            logger.debug("driver is initialized");
+
+                            insteonBinding.setIsActive(true);
+
+                            updateStatus(ThingStatus.ONLINE);
+
+                            ScheduledFuture<?> driverInitializedJob = this.driverInitializedJob;
+                            if (driverInitializedJob != null) {
+                                driverInitializedJob.cancel(true);
```

Comment:
false should be better here to not interrupt the current running thread ...

Reply:
changed now, sorry